federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -235,44 +228,40 @@ func (s *ServiceController) updateService(obj pkgruntime.Object) (pkgruntime.Obj
 	return s.federationClient.Core().Services(service.Namespace).Update(service)
 }
 
-// Run starts a background goroutine that watches for changes to federation services
-// and ensures that they have Kubernetes services created, updated or deleted appropriately.
-// federationSyncPeriod controls how often we check the federation's services to
-// ensure that the correct Kubernetes services (and associated DNS entries) exist.
-// This is only necessary to fudge over failed watches.
-// clusterSyncPeriod controls how often we check the federation's underlying clusters and
-// their Kubernetes services to ensure that matching services created independently of the Federation
-// (e.g. directly via the underlying cluster's API) are correctly accounted for.
-
-// It's an error to call Run() more than once for a given ServiceController
-// object.
+// Run starts informers, delay deliverers and workers. Workers continuously watch for events which could
+// be from federation or federated clusters and tries to reconcile the service objects from federation to
+// federated clusters.
 func (s *ServiceController) Run(workers int, stopCh <-chan struct{}) {
 	glog.Infof("Starting federation service controller")
+	defer glog.Infof("Shutting down federation service controller")
```

Comment:
Deferred statements are executed in LIFO order, so you will log "Shutting down" after you have already shut down.  I don't think that's what you want.